### ISSUE TITLE
How function YOLOV8 detection with large images

### ISSUE THREAD

**Issue by edwinsacrecinos** [ultralytics/ultralytics#8694]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello everyone, thanks for your answers.

I am work in project for detection damage in roads but i have a problems when analizy my image using my personalized model, before my questions presentation important information of project.

I have a personalized model with six class:

Pothole Low, Poth Medium, Poth High, Alligator Cracking Low, Alligator Cracking Medium, Alligator Cracking High.

My model is trained with next especifications:

images: 1200
batch: 8 
imgz: 640
image size: 2048 x 2048
epochs: 100
model: yolov8l.pt

my computer that I use to train  : i7 13k -  32 ram - 3060 -> 6GB

My questions:

1) I evaluation damages in road using large image with size is: 4098px high X 50000 px width but why detection is litle or null? Does it have to do with the size of the image?

2) My computer not support evaluation using my image of 4098px high X 50000 px width i use size in imgsz [4098,15000] instead. This has to do my bad detection?

3) what size image support yolov8 for evaluations? I have to configure something for my detection?

4) the problem is my images number or other thing?

5)  I evaluate the damage using an individual image of 2000 high x 2048 width and in this if it detects damage, that image is a piece of the large image of 50000 px

6) the problem is the size image configuration i use in evaluation or my computer?

I evaluation the large images with:

resultsDamage = searchDamageYolo(modelYolo='best.29.02.2024.pt', pathDamage=path,namePhoto=imageName,numConf=0.2,height=4098,width=15000,showDamage=True,showCon=True)

and i evaluation the small image (2000x2048 px) with:

resultsDamage = searchDamageYolo(modelYolo='best.29.02.2024.pt', pathDamage=path,namePhoto=imageName,numConf=0.2,height=640,width=640,showDamage=True,showCon=True)

if you need more information or images. you can request them. Thank you so much.

**Comment by glenn-jocher** [comment]
@edwinsacrecinos hello! Thanks for reaching out with your detailed inquiry. 🚀

When working with large images, especially those significantly larger than the dimensions used during training, detection performance can indeed be affected. Here are some concise pointers addressing your concerns:

1. **Large Image Size:** YOLOv8 can handle large images, but there's a practical limit governed by your hardware capabilities and the model's architecture. For extremely large images like yours (4098px x 50000px), it's recommended to tile the images into smaller segments closer to the training size (e.g., 2048x2048 or smaller) and then run detection on each tile. This approach helps maintain detection performance.

2. **Hardware Limitations:** The limitations you're experiencing are likely due to the immense size of the input image, which can strain your GPU memory. Using smaller tiles, as mentioned, can mitigate this.

3. **Optimal Image Size for Evaluation:** While there's no strict maximum size, sticking close to your training dimensions (2048x2048 in your case) or using the tiling method for larger images is advisable. Adjusting `imgsz` during evaluation to match or exceed training sizes can help, but be mindful of your hardware's capabilities.

4. **Detection Issues:** The number of images in your dataset and their diversity in representing the classes across different conditions are crucial. Ensure your dataset is representative of the scenarios you're evaluating. If detections work well on smaller segments but not on the large image, it's likely an issue with handling large image sizes rather than the dataset itself.

5. **Evaluating Large vs. Small Images:** Your approach of evaluating smaller images successfully indicates that your model is capable. The challenge with large images primarily lies in processing size and memory constraints.

6. **Configuration and Hardware:** Both the evaluation size configuration and hardware limitations play roles here. Opt for image sizes that your hardware can comfortably process, and consider the tiling strategy for large images.

For tiling large images, you might consider something like this (pseudo-code):

```python
# Assuming you have a function to tile your large image into smaller segments
tiles = tile_large_image(large_image_path, tile_size=(2048, 2048))
results = []
for tile in tiles:
    result = model.predict(tile, size=640)  # Adjust size as needed
    results.append(result)
# Process and combine results from all tiles
```

This approach should help you manage large images more effectively. If you have further questions or need more specific advice, feel free to ask!

**Comment by edwinsacrecinos** [comment]
Thank you so much for your answer Glenn is very complete and clear. My questions were answered and thanks for the pseudocode I will try implement in my project.

I have a news questions:

1) my team developed a script of automatic labelled for roads damages but the label encloses the damage in a square and this images with that labels they generate a bad detection when used in training but when using a polygonal labeling created with labelme the detection is good. Example of both labels:

automatic square label:
![imgC](https://github.com/ultralytics/ultralytics/assets/23007262/32da61bd-3267-4b13-8899-aaafee86b464)


polygonal label with labelme:
![imgP](https://github.com/ultralytics/ultralytics/assets/23007262/caa91243-3fa7-4047-bcf9-467bdfd48012)

**On our team we theorize that the way you are labeling affects detection. Is this correct?**

2) Next question i have. models S and N, their accuracies compared to L and X have a lot of difference in detection, example in the damages roads detection.

Thanks my team is very pleased with your answers.

**Comment by glenn-jocher** [comment]
@edwinsacrecinos, it's great to hear that you found the previous responses helpful! 😊 Let's dive into your new questions.

1) Indeed, the labeling shape can impact detection performance. YOLO models learn to predict bounding boxes as rectangles, so using square or rectangular labels during training is more straightforward. However, for objects that don't fit well into these shapes (like many road damages), polygonal labels might provide more precise contours, helping the model learn the object's shape better. While YOLOv8 directly uses rectangular bounding boxes, ensuring the labels are as tight and representative of the object shape as possible can improve detection. This might involve manually adjusting auto-labeled boxes or developing more sophisticated auto-labeling techniques that better capture the object shape. 

2) Regarding the model sizes (S, N vs. L, X), it's expected that larger models (L, X) generally perform better at the cost of increased computational requirements. They have more capacity to learn complex features, which can be particularly beneficial for challenging detection tasks like identifying various road damages. Smaller models (S, N) are faster and require less computational power, making them suitable for real-time applications or devices with limited resources, but this can come with a trade-off in accuracy.

For complex detection tasks such as road damage identification, exploring larger models might yield better results if computational resources allow. Additionally, fine-tuning the model with a carefully annotated dataset can significantly enhance detection performance across all model sizes.

Happy to help your team advance in your project, and if you have more questions or need further clarification, don't hesitate to ask!

**Comment by glenn-jocher** [comment]
@edwinsacrecinos You're welcome! I'm delighted to hear that all your questions are resolved. 😊 If anything else comes up or you need further assistance as you progress with your project, feel free to ask. Happy coding! 🚀
